src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/catalog.cpp

```diff
@@ -81,72 +82,373 @@ struct component
 };
 
 static unordered_map<wstring, shared_ptr<component>> g_types;
+static unordered_set<wstring> g_assemblies;
 
-HRESULT WinRTLoadComponent(PCWSTR manifest_path)
+HRESULT LoadFromSxSManifest(std::wstring path)
+{
+    return WinRTLoadComponentFromFilePath(path.c_str());
+}
+
+HRESULT LoadFromEmbeddedManifest(std::wstring path)
```

Comment:
There's a feature of CreateActCtxW that you can point it at a DLL or exe and it'll compute the activation context for it, including all the probing logic listed here.  Then there are APIs like https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-queryactctxw and https://docs.microsoft.com/en-us/windows/win32/api/winnt/ns-winnt-activation_context_assembly_detailed_information that let you find the list of assembly manifests used to construct the activation context, which you can then load & parse.

Activation contexts are also cacheable so you could keep a list of those somewhere.